plugins/modules/source_control/github/github_repo.py

```diff
@@ -0,0 +1,245 @@
+    - This is only makes sense when creating the new repository.
+    type: str
+    default: ''
+    required: false
+  private:
+    description:
+    - Whether the new repository should be private or not.
+    - This is only makes sense when creating the new repository.
+    type: bool
+    default: no
+    required: false
+  state:
+    description:
+    - Whether the repository should exist or not.
+    type: str
+    default: present
+    choices: [ absent, present ]
+    required: false
+  organization:
+    description:
+    - Organization for the repository.
+    - If not provided, the repository will be created in the current user profile.
+    type: str
+    required: false
+requirements:
+- PyGithub>=1.54
+notes:
+- For python3, PyGithub>=1.54 should be used.
+- "For python3.5, PyGithub==1.54 should be used. More info: https://pygithub.readthedocs.io/en/latest/changes.html#version-1-54-november-30-2020."
+- "For python2.7, PyGithub==1.45 should be used. More info: https://pygithub.readthedocs.io/en/latest/changes.html#version-1-45-december-29-2019."
+author:
+- Álvaro Torres Cogollo (@atorrescogollo)
+'''
+
+EXAMPLES = '''
+- name: Create a Github Repo
+  github_repo:
+    access_token: mytoken
+    organization: MyOrganization
+    name: myrepo
+    description: "Just for fun"
+    private: yes
+    state: present
+  register: result
+- name: Delete the repo
+  github_repo:
+    username: octocat
+    password: password
+    organization: MyOrganization
+    name: myrepo
+    state: absent
+  register: result
+'''
+
+RETURN = '''
+repo:
+  description: Repository information as JSON. See U(https://docs.github.com/en/rest/reference/repos#get-a-repository).
+  returned: success and I(state) is C(present)
+  type: dict
+'''
+
+
+import sys
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+
+GITHUB_IMP_ERR = None
+try:
+    from github import Github, GithubException
+    HAS_GITHUB_PACKAGE = True
+except Exception:
+    GITHUB_IMP_ERR = traceback.format_exc()
+    HAS_GITHUB_PACKAGE = False
+
+
+def authenticate(username=None, password=None, access_token=None):
+    if access_token:
+        return Github(base_url="https://api.github.com:443", login_or_token=access_token)
+    else:
+        return Github(base_url="https://api.github.com:443", login_or_token=username, password=password)
+
+
+def create_repo(gh, name, organization=None, private=False, description=''):
+    result = dict(
+        changed=False,
+        repo=None)
+    if organization:
+        target = gh.get_organization(organization)
+    else:
+        target = gh.get_user()
+
+    repo = None
+    try:
+        repo = target.get_repo(name=name)
+        result['repo'] = repo.raw_data
+
+        changes = {}
+        if repo.raw_data['private'] != private:
+            changes['private'] = private
+        if repo.raw_data['description'] != description:
+            changes['description'] = description
```

Comment:
Here it looks like you do use these options also on existing repositories.

Reply:
The repository needs to be edited when the private and description values are not correct. Is this a bug?

Reply:
In fact, there is an unit test for that use case: https://github.com/atorrescogollo/community.general/blob/6b24b60cdaf6ecf80507a8cea7f36e7dacd7e478/tests/unit/plugins/modules/source_control/github/test_github_repo.py#L185-L200

Am I missing something?

Reply:
This was a follow-up to my first comment, where you originally said "This is only makes sense when creating the new repository.". That would tell me as a user that these values are only used on creation, but not during update. When looking at the code, I then saw that you actually update them, and suggested to adjust that sentence to "This is only used when I(state) is C(present).". So everything's fine here :-)

Reply:
Oh I see. It was not very clear, I agree :)